### ISSUE TITLE
CustomImportOrder:  unexpected violation for extra empty line between package and import

### ISSUE THREAD

**Issue by gmshake** [checkstyle/checkstyle#6680]
```
/var/tmp $ javac CustomImportOrder.java
```

```
/var/tmp $ cat config.xml
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
 "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">

    <module name="CustomImportOrder">
      <property name="customImportOrderRules"
               value="STATIC###STANDARD_JAVA_PACKAGE"/>
      <property name="sortImportsInGroupAlphabetically" value="true"/>
    </module>

  </module>
</module>
```

```
/var/tmp $ cat CustomImportOrder.java
package com.example;


import java.util.Arrays;

class CustomImportOrder {
}
```

```
/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-8.19-all.jar \
    -c config.xml CustomImportOrder.java
Starting audit...
[ERROR] /var/tmp/CustomImportOrder.java:4: 'java.util.Arrays' should be separated 
from previous import group by one line. [CustomImportOrder]
Audit done.
Checkstyle ends with 1 errors.
```

---------------

Obviously it's a misleading error message, since there're no import groups before the only one `import java.util.Arrays;` .

I'd like to propose that we clarify the violation error message, such as `should be separated from package define by one line`

Inspired by https://github.com/checkstyle/checkstyle/issues/5304

--------------

**Comment by romani** [comment]
I do not think there should be violation, as Check should care only about separation, not size.

From documentation http://checkstyle.sourceforge.net/config_imports.html#CustomImportOrder_Properties :
>groups are separated by, at least, one blank line

So no need to change message, it simply a bug, there should be no violation.

**Comment by gmshake** [comment]
IIUC, 

> So no need to change message, it simply a bug, there should be no violation.
means that the property ``separateLineBetweenGroups`` should not check separations between package define and import groups, right?

Shall I create another issue, or attach a new pull request against this bug ?

**Comment by romani** [comment]
> separateLineBetweenGroups should not check separations between package define and import groups, right?

:) hmm , please prove by CLI that `separateLineBetweenGroups=false` make you test case to produce no violation.
By rationale "Force empty line separator between import groups." it should not affect your case at all, It should care only about BETWEEN groups.
If it affects validation - we should fix it.

You are welcome with PR.

**Comment by gmshake** [comment]
Hi @romani , 
neither ```separateLineBetweenGroups=true``` nor ```separateLineBetweenGroups=false``` there're unintended violations.
I've tested and verified that commit 914dd9994c9f108f54849a96bc35bbdd85de1053 introduced this regression. Related issue #3551 .
The commit 914dd9994c9f108f54849a96bc35bbdd85de1053 also introduced another side effect that it enforce there're no double blank lines within import group. 

The fix is simple, but I'm not sure how to describe the issue. Need your help.

**Comment by rnveach** [comment]
Fix was merged
